crates/orchestrator/src/data_storage/mod.rs

```diff
@@ -21,7 +21,10 @@ use utils::settings::Settings;
 pub trait DataStorage: Send + Sync {
     async fn get_data(&self, key: &str) -> Result<Bytes>;
     async fn put_data(&self, data: Bytes, key: &str) -> Result<()>;
-    async fn build_test_bucket(&self, bucket_name: &str) -> Result<()>;
+    async fn build_bucket(&self, bucket_name: &str) -> Result<()>;
+    async fn setup(&self, bucket_name: &str) -> Result<()> {
```

Comment:
the bucket name won't be a param to setup no? the setup function should know what buckets to create in my understanding

Reply:
It is better to have it taken as a param in case we wanted to create a new bucket.